kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/MessageFetchContext.java

```diff
@@ -398,6 +398,14 @@ private void readMessagesInternal(KafkaHeaderAndRequest fetch,
                                     responseData,
                                     ((Integer) THROTTLE_TIME_MS.defaultValue),
                                     ((FetchRequest) fetch.getRequest()).metadata().sessionId()));
+                    resultFuture.whenComplete((r, e) -> {
+                        // ATTENTION: Please note that the release of entries should be placed after
+                        // future complete operation to ensure that the data has been successfully
+                        // written to the network channel
+                        responseValues.entrySet().forEach(topicPartitionListEntry -> {
+                            topicPartitionListEntry.getValue().forEach(Entry::release);
+                        });
```

Comment:
I think there's still a race condition. The previous `whenComplete` for response future is

```java
                ctx.channel().eventLoop().execute(() -> {
                    writeAndFlushResponseToClient(channel);
                });
```

We can see `writeAndFlushResponseToClient` may not be executed immediately.